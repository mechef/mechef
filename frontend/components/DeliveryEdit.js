// @flow

import * as React from 'react';
import Rx from 'rxjs/Rx';

import Button from './Button';
import TextInput from './TextInput';
import MapWithAutoComplete from './MapWithAutoComplete';
import { MeetupObject } from '../utils/flowTypes';


type Props = {
  onCreateMeetup: (meetup: MeetupObject) => Rx.Observable,
  onUpdateMeetup: (meetup: MeetupObject) => Rx.Observable,
  onDeleteMeetup: (meetupId: string) => Rx.Observable,
  meetupList: Array<MeetupObject>,
  currentMeetupId: string,
  goBack: () => Rx.Observable,
}

type State = {
  meetup: MeetupObject,
}

class DeliveryEdit extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    const currentMeetup = props.meetupList.find(
      delivery => delivery._id === props.currentMeetupId) || {
        _id: '',
        note: '',
        meetupEndTime: '',
        meetupStartTime: '',
        meetupSaturday: false,
        meetupFriday: false,
        meetupThursday: false,
        meetupWednesday: false,
        meetupTuesday: false,
        meetupMonday: false,
        meetupSunday: false,
        meetLongitude: -1,
        meetupLatitude: -1,
        meetupAddress: '',
        type: 'meetup',
      };
    this.state = {
      meetup: currentMeetup,
    };
  }
  render() {
    const { goBack, onCreateMeetup, onUpdateMeetup, onDeleteMeetup, currentMeetupId } = this.props;
    return (
      <div className="dashboard-content">
        <p className="dashboard-content__title">Edit Delivery</p>
        <div className="editContainer">
          <div className="meetupLocation">
            <h3 className="title">Meet up location</h3>
            <p className="subtitle">Location and Time</p>
            <MapWithAutoComplete />
          </div>
          <h3 className="title">Meet up date</h3>
          <div className="meetupDate">
            <div className="smallInputContainer">
              <span className="subtitle">Select Date MM/DD/YYYY</span>
              <TextInput
                type="text"
                size="small"
              />
            </div>
            <div className="smallInputContainer">
              <span className="subtitle">Dynamic Period</span>
              <TextInput
                type="select"
                size="small"
              />
            </div>
          </div>
          <h3 className="title">Meet up time</h3>
          <div className="meetupDate">
            <div className="smallInputContainer">
              <span className="subtitle">From</span>
              <TextInput
                type="select"
                size="small"
              />
            </div>
            <div className="smallInputContainer">
              <span className="subtitle">To</span>
              <TextInput
                type="select"
                size="small"
              />
            </div>
          </div>
        </div>
        <div className="buttonGroup">
          <div>
            <Button
              buttonStyle="greenBorderOnly"
              size="small"
              onClick={() => {
                onDeleteMeetup(currentMeetupId);
                goBack();
              }}
            >
              DELETE
            </Button>
          </div>
          <div>
            <Button
              buttonStyle="greenBorderOnly"
              size="small"
              onClick={() => goBack()}
            >
              CANCEL
            </Button>
          </div>
          <div>
            <Button
              buttonStyle="primary"
              size="small"
              onClick={() => {
                if (this.state.id) {
                  // TODO: Modify to only provide updated data
                  onUpdateMeetup(this.state.meetup);
                } else {
                  onCreateMeetup(this.state.meetup);
                }
                goBack();
              }}
            >
              SAVE
            </Button>
          </div>
        </div>
        <style jsx>
          {`
            .dashboard-content {
              padding-left: 19px;
            }

            .dashboard-content__title {
              font-size: 18px;
              line-height: 1.11;
              letter-spacing: 0.5px;
              color: #4a4a4a;
            }

            .editContainer {
              margin-top: 24px;
              width: 744px;
              height: 100%;
              padding: 24px 20px;
              border-radius: 4px;
              background-color: #ffffff;
            }

            .meetupLocation {
              margin-bottom: 13px;
            }

            .title {
              margin: 0 0 16px 0;
              font-size: 16px;
              font-weight: 500;
              line-height: 1;
              letter-spacing: 0.7px;
              color: #4a4a4a;
            }

            .subtitle {
              margin: 0 0 16px 0;
              height: 14px;
              font-size: 14px;
              font-weight: 500;
              line-height: 1;
              letter-spacing: 0.6px;
              text-align: left;
              color: #9b9b9b;
            }

            .meetupDate {
              display: flex;
              justify-content: space-between;
              width: 539px;
              margin-bottom: 26px;
            }

            .smallInputContainer {
              display: flex;
              flex-direction: column;
            }

            .selectbox {
              width: 250px;
              height: 44px;
              border-radius: 4px;
              border: solid 1px #979797;
            }

            .buttonGroup {
              display: flex;
              justify-content: flex-end;
              width: 744px;
              padding-top: 30px;
            }

            .buttonGroup div {
              margin-left: 10px;
            }

            .largInput {
              width: 447px;
              height: 44px;
              opacity: 0.6;
              border-radius: 4px;
              border: solid 1px #979797;
            }
          `}
        </style>
      </div>
    );
  }
}

export default DeliveryEdit;
