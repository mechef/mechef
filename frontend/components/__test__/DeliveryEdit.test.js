import renderer from 'react-test-renderer';
import React from 'react';

import DeliveryEdit from '../DeliveryEdit';

jest.mock('react-geosuggest', () => '[MOCK]Geosuggest');

describe('Pages With Snapshot Testing', () => {
  it('snapshot DeliveryEdit Component', () => {
    global.google = {
      maps: {
        Map: jest.fn(),
      },
    };
    const component = renderer.create(
      <DeliveryEdit
        meetupList={[{
          _id: '59e5f7ffd48173ad48d8c8ab',
          note: '9F caferia',
          meetupEndTime: '',
          meetupStartTime: '',
          meetupSaturday: false,
          meetupFriday: false,
          meetupThursday: false,
          meetupWednesday: false,
          meetupTuesday: false,
          meetupMonday: false,
          meetupSunday: false,
          meetLongitude: 20,
          meetupLatitude: 120,
          meetupAddress: 'Rakuten Crimson House',
          type: 'meetup',
        }]}
        onCreateMeetup={() => {}}
        onUpdateMeetup={() => {}}
        onDeleteMeetup={() => {}}
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
