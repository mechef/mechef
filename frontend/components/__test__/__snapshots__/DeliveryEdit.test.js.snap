// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages With Snapshot Testing snapshot DeliveryEdit Component 1`] = `
<div
  className="jsx-1810684706 dashboard-content"
>
  <p
    className="jsx-1810684706 dashboard-content__title"
  >
    Edit Delivery
  </p>
  <div
    className="jsx-1810684706 editContainer"
  >
    <div
      className="jsx-1810684706 meetupLocation"
    >
      <h3
        className="jsx-1810684706 title"
      >
        Meet up location
      </h3>
      <p
        className="jsx-1810684706 subtitle"
      >
        Location and Time
      </p>
      <div
        className="jsx-2276495436 container"
      >
        <div
          className="jsx-2276495436 mapWrapper"
          id="map"
        />
        <[MOCK]Geosuggest
          onSuggestSelect={[Function]}
          placeholder="Enter meet up address"
        />
        <style>
          
          .geosuggest {
            position: relative;
            width: 511px;
            margin-bottom: 13px;
          }
          .geosuggest__input {
            outline: none;
            border-radius: 4px;
            height: 44px;
            background-color: #ffffff;
            padding-left: 18px;
            font-size: 14px;
            font-weight: 500;
            width: 511px;
            border: 2px solid transparent;
            box-shadow: 0 0 1px #3d464d;
            transition: border 0.2s, box-shadow 0.2s;
          }
          .geosuggest__input:focus {
            border-color: #3e9f40;
            box-shadow: 0 0 0 transparent;
          }
          .geosuggest__suggests {
            position: absolute;
            top: 100%;
            left: 0;
            right: 0;
            max-height: 25em;
            padding-left: 18px;
            width: 511px;
            height: 161px;
            margin-top: -1px;
            background: #fff;
            border: 2px solid #267dc0;
            border-top-width: 0;
            overflow-x: hidden;
            overflow-y: auto;
            list-style: none;
            z-index: 5;
            transition: max-height 0.2s, border 0.2s;
          }
          .geosuggest__suggests--hidden {
            max-height: 0;
            overflow: hidden;
            border-width: 0;
          }

          .geosuggest__item {
            font-size: 18px;
            font-size: 1rem;
            padding: .5em .65em;
            cursor: pointer;
          }

          .geosuggest__item:hover,
          .geosuggest__item:focus {
            background: #f5f5f5;
          }

          .geosuggest__item--active {
            background: #267dc0;
            color: #fff;
          }
          .geosuggest__item--active:hover,
          .geosuggest__item--active:focus {
            background: #ccc;
          }
          .geosuggest__item__matched-text {
            font-weight: 500;
          }

          .geosuggest__input-wrapper &gt; input::placeholder {
            line-height: 1.14;
            color: #b9b9b9;
          }

        
        </style>
      </div>
    </div>
    <h3
      className="jsx-1810684706 title"
    >
      Meet up date
    </h3>
    <div
      className="jsx-1810684706 meetupDate"
    >
      <div
        className="jsx-1810684706 smallInputContainer"
      >
        <span
          className="jsx-1810684706 subtitle"
        >
          Select Date MM/DD/YYYY
        </span>
        <div
          className="jsx-4141203587"
        >
          <input
            className="jsx-4141203587 
        textInput
        small
      "
            onChange={[Function]}
            placeholder=""
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="jsx-1810684706 smallInputContainer"
      >
        <span
          className="jsx-1810684706 subtitle"
        >
          Dynamic Period
        </span>
        <div
          className="jsx-4141203587"
        >
          <input
            className="jsx-4141203587 
        textInput
        small
      "
            onChange={[Function]}
            placeholder=""
            type="select"
            value=""
          />
        </div>
      </div>
    </div>
    <h3
      className="jsx-1810684706 title"
    >
      Meet up time
    </h3>
    <div
      className="jsx-1810684706 meetupDate"
    >
      <div
        className="jsx-1810684706 smallInputContainer"
      >
        <span
          className="jsx-1810684706 subtitle"
        >
          From
        </span>
        <div
          className="jsx-4141203587"
        >
          <input
            className="jsx-4141203587 
        textInput
        small
      "
            onChange={[Function]}
            placeholder=""
            type="select"
            value=""
          />
        </div>
      </div>
      <div
        className="jsx-1810684706 smallInputContainer"
      >
        <span
          className="jsx-1810684706 subtitle"
        >
          To
        </span>
        <div
          className="jsx-4141203587"
        >
          <input
            className="jsx-4141203587 
        textInput
        small
      "
            onChange={[Function]}
            placeholder=""
            type="select"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="jsx-1810684706 buttonGroup"
  >
    <div
      className="jsx-1810684706"
    >
      <button
        className="jsx-675350272 
      button
      greenBorderOnly
      small
    "
        onClick={[Function]}
        type="button"
      >
        DELETE
      </button>
    </div>
    <div
      className="jsx-1810684706"
    >
      <button
        className="jsx-675350272 
      button
      greenBorderOnly
      small
    "
        onClick={[Function]}
        type="button"
      >
        CANCEL
      </button>
    </div>
    <div
      className="jsx-1810684706"
    >
      <button
        className="jsx-675350272 
      button
      primary
      small
    "
        onClick={[Function]}
        type="button"
      >
        SAVE
      </button>
    </div>
  </div>
</div>
`;
